components/series/SeriesTable.vue

```diff
@@ -353,17 +365,71 @@ export default class SeriesTable extends mixins(PrefixMixin) {
       data: { collectionEntities },
     } = collections
 
+    const defaultBuyEvents = getDateArray(lastmonthDate, today).reduce(
+      (res, date) => {
+        res[date] = 0
+        return res
+      },
+      {}
+    )
+
+    const axisLize = (obj = {}): BuyHistory => ({
+      xAxisList: Object.keys(obj),
+      yAxisList: Object.values(obj),
+    })
+
+    const ids = collectionEntities.map((c: Collection) => c.id)
+
+    const buyEvents = (await this.fetchCollectionEvents(ids))
+      .map((e) => ({
+        ...e.nft.collection,
+        timestamp: e.timestamp.replace(/(T.*?$)/g, ''),
+      }))
+      .reduce((res, e) => {
+        const { id, timestamp: ts } = e
+        if (!res[id]) {
+          res[id] = Object.assign({}, defaultBuyEvents)
+        }
+        res[id][ts] += 1
```

Comment:
ooof

Reply:
Do you mind expanding the review comment?

Some thoughts:

Currently, it's quite dirty for the part of initialization and code related to `this.fetchCollectionEvents`, maybe we can hold that when we finished the resolver, then the transformation of fetched data would be much neater or not needed anymore. I can come back to refactor that.

Reply:
I am willing to work on that resolver as well

get a keen desire 😄 to write code about that.